src/utils.rs

```diff
@@ -304,20 +304,36 @@ pub(crate) mod testing {
     fn init_rng() -> StdRng {
         let mut seeder = OsRng;
         let seed = seeder.gen();
+        eprintln!(
+            "Test Failed. To recreate the randomness used, use init_testing_with_seed() with the following seed:"
+        );
         eprintln!("seed: {seed:?}");
         StdRng::from_seed(seed)
     }
 
+    /// Returns an rng with a manually-specified seed to be used for debugging.
+    #[allow(unused)]
+    fn init_rng_with_seed(seed: [u8; 32]) -> StdRng {
+        StdRng::from_seed(seed)
+    }
```

Comment:
suggestion: this doesn't have to be a separate function, I'd recommend just creating the RNG directly in `init_testing_with_seed`.